test/Microsoft.Azure.Relay.UnitTests/HybridRequestTests.cs

```diff
@@ -780,6 +779,89 @@ async Task Verbs()
             }
         }
 
+        [Fact, DisplayTestMethodName]
+        async Task SmallRequestSmallResponseParallel()
+        {            
+            var cts = new CancellationTokenSource(TimeSpan.FromSeconds(180));
+            HybridConnectionListener listener = null;
+            try
+            {
+                var endpointTestType = EndpointTestType.Unauthenticated;
+                listener = this.GetHybridConnectionListener(endpointTestType);
+                RelayConnectionStringBuilder connectionString = GetConnectionStringBuilder(endpointTestType);
+                Uri endpointUri = new UriBuilder(connectionString.Endpoint)
+                {
+                    Scheme = Uri.UriSchemeHttps, Path = connectionString.EntityPath
+                }.Uri;
+
+                byte[] responseBytes = Encoding.UTF8.GetBytes("{ \"a\" : true }");
+                listener.RequestHandler = async (context) =>
+                {
+                    try
+                    {
+                        context.Response.StatusCode = HttpStatusCode.OK;
+                        context.Response.OutputStream.Write(responseBytes, 0, responseBytes.Length);
+                        await context.Response.CloseAsync();
+                    }
+                    catch (Exception e)
+                    {
+                        TestUtility.Log($"ERROR: {e}");
+                        throw;
+                    }
+                };
+
+                TestUtility.Log($"Opening {listener}");
+                await listener.OpenAsync(cts.Token);
+
+                string sasToken = string.Empty;
+                var tokenProvider = CreateTokenProvider(connectionString);
+                if (tokenProvider != null)
+                {
+                    sasToken = (await tokenProvider.GetTokenAsync(endpointUri.AbsoluteUri, TimeSpan.FromMinutes(30))).TokenString;
+                }
+
+                const int ParallelCount = 5; // 50 * 400 gets good throughput but requires warmup time to ramp up the threadpool
+                const int MessageCount = 10; // 400;
```

Comment:
>10 [](start = 41, length = 2)

So this will send 5 * 10 = 50 requests messages?

Reply:
that is correct. Some better variable names could help. "SenderCount" and "MessagesPerSender" maybe.

---
In reply to: [315426868](https://github.com/Azure/azure-relay-dotnet/pull/112#discussion_r315426868) [](ancestors = 315426868)

Reply:
Renamed and clarified in iteration 2, lines 368-370.

---
In reply to: [315428177](https://github.com/Azure/azure-relay-dotnet/pull/112#discussion_r315428177) [](ancestors = 315428177,315426868)